.github/workflows/sonar-cloud-static-analysis.yml

```diff
@@ -98,19 +98,26 @@ jobs:
 
       - name: Create build directory
         run: |
-          mkdir -p ${{ github.workspace }}/Build
-          mkdir -p ${{ github.workspace }}/Build/UCD
-          mkdir -p ${{ github.workspace }}/Build/CLDR
+          mkdir -p ${{ github.workspace }}/Build/${{ env.SONAR_ANALYSIS_ARCH }}/UCD
+          mkdir -p ${{ github.workspace }}/Build/${{ env.SONAR_ANALYSIS_ARCH }}/CLDR
 
       - name: Create build environment
-        working-directory: ${{ github.workspace }}/Build
-        run: cmake .. -GNinja -DSERENITY_ARCH=i686 -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DENABLE_PCI_IDS_DOWNLOAD=OFF -DENABLE_USB_IDS_DOWNLOAD=OFF -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10
-
+        working-directory: ${{ github.workspace }}
+        run: |
+          cmake -S Meta/CMake/Superbuild -B Build/superbuild -GNinja \
+            -DSERENITY_ARCH=${{ env.SONAR_ANALYSIS_ARCH }} \
+            -DSERENITY_TOOLCHAIN=GNU \
+            -DCMAKE_C_COMPILER=gcc-10 \
+            -DCMAKE_CXX_COMPILER=g++-10 \
+            -DENABLE_PCI_IDS_DOWNLOAD=OFF \
+            -DENABLE_USB_IDS_DOWNLOAD=OFF
 
       - name: Build generated sources so they are available for analysis.
-        working-directory: ${{ github.workspace }}/Build
+        working-directory: ${{ github.workspace }}
         run: |
-          ninja all_generated
+          ninja -C Build/superbuild serenity-configure
```

Comment:
Can you add a comment about what's going on here, I'm sure this works, but I'm lost as to how the Build/{{ env.SONAR_ANALYSIS_ARCH }} gets created.

Reply:
It gets created/populated via the configure step of the serenity ExternalProject in serenity-configure from the super build.

```
set(SERENITY_BUILD_DIR "${PROJECT_BINARY_DIR}/../${SERENITY_ARCH}${SERENITY_BUILD_DIR_SUFFIX}")
....
ExternalProject_Add(
    serenity
    SOURCE_DIR "${SERENITY_SOURCE_DIR}"
    BINARY_DIR "${SERENITY_BUILD_DIR}"
    ...
    STEP_TARGETS configure install
    ...
)
```
but yah I can slap a comment on it.